// graphql-playground-middleware-hapi/dist/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
var graphql_playground_html_1 = require("graphql-playground-html");
var plugin = {
    name: 'graphql-playground',
    register: function (server, options) {
        if (arguments.length !== 2) {
            throw new Error("Playground middleware expects exactly 2 arguments, got " + arguments.length);
        }
        var path = options.path, _a = options.route, config = _a === void 0 ? {} : _a, rest = __rest(options, ["path", "route"]);
        var middlewareOptions = __assign({}, rest);
        server.route({
            method: 'GET',
            path: path,
            config: config,
            handler: function (_request, h) {
                return h.response(graphql_playground_html_1.renderPlaygroundPage(middlewareOptions)).type('text/html');
            },
        });
    },
};
exports.default = plugin;
